passes/cmds/select.cc

```diff
@@ -1581,15 +1578,15 @@ struct CdPass : public Pass {
 
 		std::string modname = RTLIL::escape_id(args[1]);
 
-		if (design->modules_.count(modname) == 0 && !design->selected_active_module.empty()) {
-			RTLIL::Module *module = NULL;
-			if (design->modules_.count(design->selected_active_module) > 0)
-				module = design->modules_.at(design->selected_active_module);
-			if (module != NULL && module->cells_.count(modname) > 0)
-				modname = module->cells_.at(modname)->type.str();
+		if (!design->has(modname) == 0 && !design->selected_active_module.empty()) {
+			RTLIL::Module *module = nullptr;
+			if (design->has(design->selected_active_module) > 0)
+				module = design->module(design->selected_active_module);
+			if (module != nullptr && module->cells().contains(modname))
```

Comment:
I don't think we need a `ObjRange::contains()` method; one can look at whether the `Module::cell()` returns a null pointer or not.

Reply:
Got it.